File: src/main/java/net/thedragonteam/armorplus/items/books/ItemAPBook.java
package net.thedragonteam.armorplus.items.books;

import net.minecraft.client.Minecraft;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import net.minecraft.util.ActionResult;
import net.minecraft.util.EnumActionResult;
import net.minecraft.util.EnumHand;
import net.minecraft.world.World;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import net.thedragonteam.armorplus.client.gui.GuiArmorPlusInfo;
import net.thedragonteam.armorplus.items.base.ItemBase;

/**
 * @author Sokratis Fotkatzikis - TheDragonTeam
 **/
public class ItemAPBook extends ItemBase {

    public ItemAPBook() {
        super("book");
    }

    @Override
    @SideOnly(Side.CLIENT)
    public ActionResult<ItemStack> onItemRightClick(World worldIn, EntityPlayer player, EnumHand hand) {
        if (!player.getHeldItem(hand).isEmpty()) {
            Minecraft.getMinecraft().displayGuiScreen(new GuiArmorPlusInfo());
        }
        return new ActionResult<>(EnumActionResult.PASS, player.getHeldItem(hand));
    }

}

File: src/main/java/net/thedragonteam/armorplus/items/books/ItemAPBook.kt
/*
 * Copyright (c) TheDragonTeam 2016-2017.
 */

package net.thedragonteam.armorplus.items.books

import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.ItemStack
import net.minecraft.util.ActionResult
import net.minecraft.util.EnumActionResult
import net.minecraft.util.EnumHand
import net.minecraft.world.World
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly
import net.thedragonteam.armorplus.items.base.BaseItem

class ItemAPBook : BaseItem("book") {

    @SideOnly(Side.CLIENT)
    override fun onItemRightClick(world: World?, player: EntityPlayer, hand: EnumHand): ActionResult<ItemStack> {
        //     if (!player.getHeldItem(hand).isEmpty)
        //         Minecraft.getMinecraft().displayGuiScreen(GuiArmorPlusInfo())
        return ActionResult(EnumActionResult.PASS, player.getHeldItem(hand))
    }

}